File: Practice/src/introduction_coding_tests/ClosestNumber.java
package introduction_coding_tests;

import java.util.Arrays;

public class ClosestNumber {
//	가까운 수
	public int solution(int[] array, int n) {
		return Arrays.stream(array).reduce(200, (p, c) -> {
			int a = Math.abs(n - p);
			int b = Math.abs(n - c);
			if (a == b && p < c || a < b) {
				return p;
			} else {
				return c;
			}
		});
	}
}

File: Practice/src/introduction_coding_tests/ClosestNumber.js
function solution(array, n) {
    return array.reduce((p, c) =>
        Math.abs(n - p) < Math.abs(n - c) ||
        (Math.abs(n - p) == Math.abs(n - c) && p < c)
            ? p
            : c
    );
}
